### PR TITLE
Added: close button in the tag modal (#85zrvzfqv)

### LINKED ISSUE #193: Add close button to tag modal
## What is the motivation for adding/enhancing this feature?

The "add tag" modal doesn't have a dedicated button for closing it which makes it confusing for the user when they are don't selecting their tags and want to exit the screen. Adding screenshot for proposed UI of closing modal.

Screenshot 2023-02-23 at 4 23 28 PM

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues
 

 Closes #193 

 ### Short Description and Why It's Useful
 
Added a close/check button to save the applied tag filters in the modal. Previously, the tags were applied directly on checking/selecting but now the changes will take effect after clicking the check button.

 ### Screenshots of Visual Changes before/after (If There Are Any)
 
[Screencast from 10-04-23 05:45:47 PM IST.webm](https://user-images.githubusercontent.com/59442907/230899959-1a39b5c7-44f4-4251-83fa-61d6e71f39d9.webm)

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/components/ProductFilterModal.vue [hunk 1]
```diff
@@ -87,6 +97,9 @@ export default defineComponent({
     })
   },
   props: ["label", "facetToSelect", "searchfield", 'type'],
+  mounted() {
+    this.selectedTags = JSON.parse(JSON.stringify(this.appliedFilters[this.type][this.searchfield])).list;
```
Comment (adityasharma7) [inline src/components/ProductFilterModal.vue:101]: This should not be tag specific

File: src/store/modules/product/actions.ts [hunk 2]
```diff
@@ -60,7 +60,11 @@ const actions: ActionTree<ProductState, RootState> = {
   async updateAppliedFilters({ commit, state, dispatch }, payload) {
     const value = payload.value
     const appliedFilters = JSON.parse(JSON.stringify((state.appliedFilters as any)[payload.type][payload.id]))
-    appliedFilters.list.includes(value) ? appliedFilters.list.splice(appliedFilters.list.indexOf(value), 1) : appliedFilters.list.push(value)
+    if(!Array.isArray(value)) {
+      appliedFilters.list.includes(value) ? appliedFilters.list.splice(appliedFilters.list.indexOf(value), 1) : appliedFilters.list.push(value)
+    } else {
+      appliedFilters.list = value
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:66]: We could set value in all the cases, checking if exist could be done by callee

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts [hunk 3]
```diff
@@ -58,19 +58,14 @@ const actions: ActionTree<ProductState, RootState> = {
...
-    appliedFilters.list.includes(value) ? appliedFilters.list.splice(appliedFilters.list.indexOf(value), 1) : appliedFilters.list.push(value)
-    commit(types.PRODUCT_FILTER_UPDATED, {id: payload.id, type: payload.type, value: appliedFilters})
+    commit(types.PRODUCT_FILTER_UPDATED, { id: payload.id, type: payload.type, value: payload.value })
     dispatch('updateQuery')
   },
 
   async updateAppliedFilterOperator({ commit, state, dispatch }, payload) {
-    const appliedFilters = JSON.parse(JSON.stringify((state.appliedFilters as any)[payload.type][payload.id]))
-    appliedFilters.operator = payload.value;
-    commit(types.PRODUCT_FILTER_UPDATED, {id: payload.id, type: payload.type, value: appliedFilters})
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:71]: Why this change is needed?
This action should only allow operator change. With this change, this action is updating Applied Filters like the above action with an exception of querying on condition

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 3 above)
Comment (k2maan) [reply src/store/modules/product/actions.ts]: Updated sir

**Review by adityasharma7 — APPROVED** [review]
